### PR TITLE
doc/dev: Elaborate on boost .deb creation

### PR DESCRIPTION
**Pull request by djgalloway**
Signed-off-by: David Galloway <dgallowa@redhat.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by anthonyeleven — APPROVED** [review]
Lgtm

File: doc/dev/continuous-integration.rst [hunk 1]
```diff
@@ -155,14 +155,25 @@ libboost
     ``ceph-libboost-*``, and they are instead installed into ``/opt/ceph``, so
     they don't interfere with the official ``libboost`` packages shipped by
     distro. Its build scripts are hosted at https://github.com/ceph/ceph-boost.
+    See https://github.com/ceph/ceph-boost/commit/2a8ae02932b2a1fd6a68072da8ca0df2b99b805c
```
Comment (tchaikov) [inline doc/dev/continuous-integration.rst:158]: how about adding a note on how to use `dch` to add a changelog item, something like. 
```rst

to add a boilerplate changelog entry:

.. prompt:: bash $

   dch -v 1.80.0-1.1

```

**Review by tchaikov — APPROVED** [review]

File: doc/dev/continuous-integration.rst (hunk 1 above)
Comment (djgalloway) [reply doc/dev/continuous-integration.rst:158]: Shoot, I just missed this.  I actually edited the changelog manually :grimacing:

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/47415 - Quincy backport

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/47416 - Pacific backport

**Pull request merged**